src/server/scripts/ExilesReach/zone_exiles_reach.cpp

```diff
@@ -5920,6 +5925,886 @@ class quest_resizing_the_situation_horde : public quest_resizing_the_situation
+            return;
+
+        target = survivor;
+    }
+
+    void Register() override
+    {
+        OnObjectTargetSelect += SpellObjectTargetSelectFn(spell_summon_darkmaul_plains_questgivers_q55879::SelectTarget, EFFECT_0, TARGET_DEST_NEARBY_ENTRY_OR_DB);
+    }
+};
+
+enum SpellRidingGiantBoar
+{
+    SPELL_RIDING_GIANT_BOAR_305068 = 305068,
+    SPELL_RIDING_GIANT_BOAR_321670 = 321670
+};
+
+// Spell 173426 - Riding Giant Boar
+class spell_riding_giant_boar_q55879 : public AuraScript
+{
+    void OnAuraRemoveHandler(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Player* player = Object::ToPlayer(GetCaster());
+        if (!player)
+            return;
+
+        player->RemoveAura(SPELL_RIDING_GIANT_BOAR_305068);
+        player->RemoveAura(SPELL_RIDING_GIANT_BOAR_321670);
+        player->CastSpell(player, SPELL_UPDATE_PHASE_SHIFT);
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_riding_giant_boar_q55879::OnAuraRemoveHandler, EFFECT_0, SPELL_AURA_CONTROL_VEHICLE, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+enum SpellKnockbackHint
+{
+    ACTOR_ALLIANCE_CAPTAIN = 71372
+};
+
+// Spell 305742 Resizer Hit
+class spell_knockback_hint_q56034 : public SpellScript
+{
+    void HandleLaunch(SpellEffIndex effIndex)
+    {
+        PreventHitDefaultEffect(effIndex);
+    }
+
+    void HandleEffect(SpellEffIndex effIndex)
+    {
+        Player* player = Object::ToPlayer(GetCaster());
+        if (!player)
+            return;
+
+        Creature* gasrrick = player->FindNearestCreatureWithOptions(10.0f, { .CreatureId = NPC_ALLIANCE_CAPTAIN, .OwnerGuid = player->GetGUID() });
+        if (!gasrrick)
+            return;
+
+        Conversation* conversation = Conversation::CreateConversation(GetSpellInfo()->GetEffect(effIndex).MiscValue, player, *player, player->GetGUID(), nullptr, false);
+        if (!conversation)
+            return;
+
+        conversation->AddActor(ACTOR_ALLIANCE_CAPTAIN, 0, gasrrick->GetGUID());
+        conversation->Start();
+    }
+
+    void Register() override
+    {
+        OnEffectLaunchTarget += SpellEffectFn(spell_knockback_hint_q56034::HandleLaunch, EFFECT_0, SPELL_EFFECT_CREATE_PRIVATE_CONVERSATION);
+        OnEffectHitTarget += SpellEffectFn(spell_knockback_hint_q56034::HandleEffect, EFFECT_0, SPELL_EFFECT_CREATE_PRIVATE_CONVERSATION);
+    }
+};
+
+enum CaptainGarrarkGiantBoar
+{
+    ACTION_EXIT_BOAR                                       = 1,
+
+    ACTOR_HENRY_GARRICK_PRISONER                           = 78493,
+
+    CONVERSATION_CAPTAIN_GARRICK_RIDE_BOAR_QUEST_ACCEPT    = 12090,
+
+    CONVERSATION_CAPTAIN_GARRICK_RIDE_BOAR_QUEST_HENRY     = 15615,
+    CONVERSATION_CAPTAIN_GARRICK_RIDE_BOAR_QUEST_EXIT      = 12092,
+
+    EVENT_CAPTAIN_GARRICK_RIDE_BOAR                        = 1,
+    EVENT_CAPTAIN_GARRICK_RIDE_BOAR_CHECK_OWNER            = 2,
+    EVENT_CAPTAIN_GARRICK_RIDE_BOAR_TALK_TO_HENRY          = 3,
```

Comment:
use some more generic-ish names for the events used by both leaders something along `EVENT_LEADER_RIDE_BOAR_TALK_TO_SON` (same with other shared events)